### ISSUE TITLE
Roll without replacement

### ISSUE THREAD

**Issue by kwollam** [colinbate/rolltable-requester#8]
When a table is set to roll without replacement there are two issues:
1) this requester does not honor locked items, it simply provides a blank output.  When actually rolling on the table foundry rerolls until an acceptable entry is rolled.

![image](https://github.com/colinbate/rolltable-requester/assets/156491382/5c8ef7f8-9b60-4319-8f82-113456e36e3f)

2) does not mark rolled entries as locked for reroll so they can be triggered multiple times. should switch them to locked. 
![image](https://github.com/colinbate/rolltable-requester/assets/156491382/26c1cebc-ea3b-4ed2-b2c5-8ef4cf61bd08)

verified with find my culprit that no other module is interfering with this functionality. with all other modules deactivated the issue persists.

**Comment by TomKeegasi** [comment]
Yeah. Noticed this too while tryig to play The Quiet Year. Makes this somewhat useless for this use-case in this state.
